packages/react-client/src/ReactFlightClient.js

```diff
@@ -854,18 +900,90 @@ export function processBinaryChunk(
   response: Response,
   chunk: Uint8Array,
 ): void {
-  const stringDecoder = response._stringDecoder;
-  let linebreak = chunk.indexOf(10); // newline
-  while (linebreak > -1) {
-    const fullrow =
-      response._partialRow +
-      readFinalStringChunk(stringDecoder, chunk.subarray(0, linebreak));
-    processFullRow(response, fullrow);
-    response._partialRow = '';
-    chunk = chunk.subarray(linebreak + 1);
-    linebreak = chunk.indexOf(10); // newline
+  let i = 0;
+  while (i < chunk.length) {
```

Comment:
same for chunk.length. It might inline it, but it might not
```suggestion
const chunkLength = chunk.length;
  while (i < chunkLength) {
```

you can see an example in JSC's JS builtins where they copy the length to a local variable

https://github.com/oven-sh/WebKit/blob/autobuild-b2f1006a06f81bc860c89dd4c7cec3e7117c4c4c/Source/JavaScriptCore/builtins/ArrayPrototype.js#L252-L289

Reply:
This is not actually a particularly hot loop. There's not that many "rows" and it doesn't read the whole chunk. It skips past most of it. Some chunks may just be completely skipped. That said, that would be aligned with our general style guide and probably smaller code so I'll do it anyway.